pkg/noderesourcetopology/score.go

```diff
@@ -0,0 +1,117 @@
+package noderesourcetopology
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+const (
+	defaultWeight = int64(1)
+)
+
+type scoreStrategy func(v1.ResourceList, v1.ResourceList, resourceToWeightMap) int64
+
+// resourceToWeightMap contains resource name and weight.
+type resourceToWeightMap map[v1.ResourceName]int64
+
+// weight return the weight of the resource and defaultWeight if weight not specified
+func (rw *resourceToWeightMap) weight(r v1.ResourceName) int64 {
```

Comment:
As `map` is already a pointer-based structure, we can just use the concrete value as the method receiver.

```suggestion
func (rw resourceToWeightMap) weight(r v1.ResourceName) int64 {
```

So on L25, we just call `w, ok := rw[r]`